api/rest/swagger.yaml

```diff
@@ -0,0 +1,796 @@
+  /version:
+    get:
+      summary: Get Version of Cluster
+      tags:
+        - cluster
+      description: Get the current version of the IPFS Cluster.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  Version:
+                    type: string
+              example: {"Version":"0.10.0"}
+              
+  /add:
+    post:
+      summary: Add a File
+      parameters:
+        - $ref: '#/components/parameters/pinNameParam'
+        - $ref: '#/components/parameters/pinReplicationMinParam'
+        - $ref: '#/components/parameters/pinReplicationMaxParam'
+        - $ref: '#/components/parameters/pinShardSize'
+        - $ref: '#/components/parameters/pinMetaParam'
+        - $ref: '#/components/parameters/hash'
+        - $ref: '#/components/parameters/recursive'
+        - $ref: '#/components/parameters/layout'
+        - $ref: '#/components/parameters/rawLeaves'
+        - $ref: '#/components/parameters/hidden'
+        - $ref: '#/components/parameters/wrap'
+        - $ref: '#/components/parameters/shard'
+        - $ref: '#/components/parameters/progress'
+        - $ref: '#/components/parameters/cidVersion'
+        - $ref: '#/components/parameters/streamChannels'
+        - $ref: '#/components/parameters/noCopy'
+        - $ref: '#/components/parameters/chunker'
+      tags:
+        - add
+      description: Add a file to IPFS via IPFS Cluster
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: string
+              format: binary
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  size:
+                    type: integer
+                  name:
+                    type: string
+                  cid:
```

Comment:
maybe worth defining this part of the schema separately for all the places in which a CID is returned (I'm not sure swagger will allow nesting schemas like that)